Auto update parquets

nup002 opened the pull request:
This pull request adds new functionality:

- Optional direct update of parquet files
Parquet files can be updated directly without the need for intermediate CSV files. This makes it much more convenient for users who have downloaded the parquet files from kaggle and wish to keep the up to date. 
- Improved output
A progressbar is displayed during the fetching of candles, showing the user how many candles will be retrieved, and how long it is expected to take.
![image](https://user-images.githubusercontent.com/15835242/110095597-5f787780-7d9d-11eb-9599-1d2b934d0013.png)

- Additional user settings
The user can set global variables on the top of Main.py to control how the script behaves. 
![image](https://user-images.githubusercontent.com/15835242/110095710-7ae38280-7d9d-11eb-99c8-74de7fc2fa74.png)

The code has also been refactored in order to more easily implement the changes listed above. The functionality of the code has not been changed, and the script can be used just as before this pull request.

yo2x committed:
[ADD] Added support for catching other type of connection errors

whirlwide committed:
Merge pull request #17 from yo2x/feature/Catch_connection_errors

[ADD] Added support for catching other types of connection errors

yo2x committed:
[FIX] added a timeout of 30 seconds by default

yo2x committed:
[FIX] some typo

whirlwide committed:
Merge pull request #19 from yo2x/feature/Catch_connection_errors

[FIX] added a timeout of 30 seconds by default

nup002 committed:
Refactored code to allow for further development.
Added progressbar to data retrieval.

nup002 committed:
Added additional user settings to top of Main.py.
Removed unecessary prints.
Refactored code further.
Implemented direct-to-parquet candles update.

nup002 committed:
Improved documentation.
Compressed (parquet) and data (csv) folders specified with top level settings.
Removed code related to development.

nup002 committed:
When running in pycharm output is now correctly formatted.

nup002 committed:
Fixed issue where, rarely, the last received maximum candle time exceeded the local time and would cause a progressbar exception.

nup002 commented:
I have noticed a few minor issues that I will attempt to correct. Please wait with the merge.

jorijnsmit commented:
Wooooow this has been on my list for so long (#13), really appreciate this! I wouldn't even mind getting rid of the csv files completely—they just take up space. I will wait for your final PR and dig through the code. Impression so far is awesome!

nup002 committed:
More robust progress bar.
New setting: Optional skipping of inactive pairs.
User options are printed when program runs.

nup002 commented:
All right, it's good to go!

nup002 commented:
Happy you like it. A few more issues to iron out.

nup002 commented:
Okay, it's ready on my end. There were no additional issues that had to be fixed.

gosuto-inzasheru commented on main.py:
```diff
@@ -76,121 +89,246 @@ def get_batch(symbol, interval='1m', start_time=0, limit=1000):
+        if bar is not None:
+            time_covered = datetime.fromtimestamp(last_timestamp / 1000) - start_datetime
+            minutes_covered = int(time_covered.total_seconds()/60)
+            bar.max_value = max(int((datetime.now() - start_datetime).total_seconds()/60), minutes_covered)
```
@nup002 can I ask what you are doing here? It seems to introduce a bug where the max of the bar can become higher than what is communicated in line 154 (`total_minutes_of_data`).

Wouldn't it be better for consistency to stick to `max(bar.max_value, minutes_covered)`?

nup002 replied:
Yes, `max(bar.max_value, minutes_covered)` is a much more elegant solution.
